enhancements/rhcos/static-networking-enhancements.md

```diff
@@ -0,0 +1,106 @@
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+This enhancement describes improvements to how static networking is configured on Fedora CoreOS (FCOS) and RHEL CoreOS (RHCOS).  It covers improving the configuration of static networking for bare metal and VMWare hosts.
+
+## Motivation
+Many customer environments require static networking and do not allow any type of DHCP server. This is most prominent in VMware environments where they struggle with the OVA images and are resisting the bare metal installer (via ISO) approach. This is also a challenge in bare metal environments due to an unfriendly and time sensitive interactive flow (i.e. catching the `grub` prompt). Due to the intersections of Ignition, Platform ID, and active networking in the `initrd`, many of the current RHCOS images append `ip=dhcp,dhcp6` and catching the bootloader on the console is not a pleasant experience. 
+
+This is basic networking configuration and users assume basic configurations like this are possible. As RHCOS requires a functional networking config early in the boot process, this eliminates many of the existing mechanisms customers use to handle the assigning of static networking configs. Some of the OpenShift personas we target are not aware of `dracut` arguments and it’s not a great experience to expect them to understand very low-level details of how the OS works.
+
+### Goals
+  - Provide a guided configuration of networking and Ignition source information
+  - Provide tooling to create customized ISOs that include unique Ignition configurations
+  - Provide support for a VMWare "backchannel" to provide network configuration and Ignition source information
+
+### Non-Goals
+  - Providing a fully automated solution for providing static network configuration to a fleet of machines
+
+## Proposal
+For the bare metal use case, we are proposing the use of the new [live ISO](https://github.com/openshift/enhancements/pull/210) to provide an interactive install environment where users can configure their networking parameters and have them persisted into the installed system.  In this environment, users will be able to invoke a TUI that will allow them to configure networking and Ignition source.  We believe this functionality will initially be most useful for users deploying Openshift Container Platform (OCP) onto bare metal using the user provisioned infrastructure (UPI) workflow.
+
+Additionally, the new live ISO will provide the ability to embed the networking configuration via an Ignition config, resulting in a custom ISO that can be used to install to a host in an automated fashion.  This would be done via the new `coreos-installer embed iso` functionality.
```

Comment:
Does this have to be tied to the live ISO? If you look at how cloud-init is structured, they have support for preflight actions that happen before networking. I really believe ignition needs something similar.

This is problematic for us in metal-ipi because the stub ignition the installer generates has a network dependency (an append with an HTTP URL), but in some customer environments to even get access to the network you need to do some configuration.

I'd like some kind of solution where ignition could do things both prior and after networking is available, e.g. deferring things like appends with network sources (or merges in V3 of ignition), but allowing us to do some networking configuration before NetworkManager is up.

Reply:
Technically...it doesn't and if implemented in Ignition it wouldn't be tied to the live ISO.  That said, I think you really want to use the Live ISO because once osmet lands for RHCOS it will be a "one stop" model for booting ephemerally and then installing without requiring an additional metal artifact.

Reply:
How do you see this working in the IPI install methods for the environments targeted by this enhancement (metal, vsphere) ?

Reply:
Stephen took the networking sub-thread to https://github.com/coreos/ignition/issues/979

Reply:
Does this enhancement need to be updated to reflect discussions that have happened in the meantime there?